runtime/lua/vim/treesitter.lua

```diff
@@ -391,6 +396,7 @@ function M.get_node(opts)
 
   assert(row >= 0 and col >= 0, 'Invalid position: row and col must be non-negative')
 
+  ---@type Range4
   local ts_range = { row, col, row, col }
```

Comment:
Then, these lines are also wrong? Or, we are just abusing 0-length ranges?

Reply:
Yes this is wrong. Needs to be `col + 1`.

Reply:
I see, thanks for confirming. Will fix this in another follow-up PR.

Reply:
Just fixed this together in the rebased commit.